In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

The file I want to use is "NASA_Labs_Facilities.xlsx". I will convert it first to ".csv"

In [6]:
file_xls = pd.read_excel('NASA_Labs_Facilities.xlsx', 'Sheet1' )
file_xls.to_csv('NASA_Labs.csv', encoding = 'Latin-1')

In [12]:
datafile = pd.read_csv('NASA_Labs.csv', encoding = 'cp1252')

In [14]:
#Number of rows
len(datafile)

440

In [16]:
#Number of columns
len(datafile.columns)

18

The file 'NASA_Labs.csv' contains 440 rows and  18 columns

In [17]:
datafile

Unnamed: 0 Unnamed: 1                                         Unnamed: 2  \
0       Agency        NaN                                             Center   
1            1        DOD                      Air Force Research Laboratory   
2            2        DOD                      Air Force Research Laboratory   
3            3        DOD                      Air Force Research Laboratory   
4            4        DOD                      Air Force Research Laboratory   
5            5        DOD              Arnold Engineering Development Center   
6            6        DOD              Arnold Engineering Development Center   
7            7        DOD              Arnold Engineering Development Center   
8            8        DOE              Pacific Northwest National Laboratory   
9            9        DOE              Pacific Northwest National Laboratory   
10          10        DOE              Pacific Northwest National Laboratory   
11          11        DOE              Pacific Northwest National Laboratory   
12          12        DOE              Pacific Northwest National Laboratory   
13          13        DOE              Pacific Northwest National Laboratory   
14          14        DOE              Pacific Northwest National Laboratory   
15          15        DOE              Pacific Northwest National Laboratory   
16          16        DOE              Pacific Northwest National Laboratory   
17          17        DOE              Pacific Northwest National Laboratory   
18          18        DOE              Pacific Northwest National Laboratory   
19          19        DOE              Pacific Northwest National Laboratory   
20          20   Intelsat                                   Intelsat General   
21          21   Intelsat                                   Intelsat General   
22          22   Intelsat                                   Intelsat General   
23          23   Intelsat                                   Intelsat General   
24          24   Intelsat                                   Intelsat General   
25          25   Intelsat                                   Intelsat General   
26          26   Intelsat                                   Intelsat General   
27          27   Intelsat                                   Intelsat General   
28          28   Intelsat                                   Intelsat General   
29          29   Intelsat                                   Intelsat General   
..         ...        ...                                                ...   
410        410     NASA 2                               Stennis Space Center   
411        411     NASA 2                               Stennis Space Center   
412        412     NASA 2                               Stennis Space Center   
413        413     NASA 2                               Stennis Space Center   
414        414     NASA 2                               Stennis Space Center   
415        415     NASA 2                               Stennis Space Center   
416        416     NASA 2                               Stennis Space Center   
417        417     NASA 2                               Stennis Space Center   
418        418     NASA 2                               Stennis Space Center   
419        419     NASA 2                               Stennis Space Center   
420        420     NASA 2                               Stennis Space Center   
421        421     NASA 2                               Stennis Space Center   
422        422     NASA 2                               Stennis Space Center   
423        423     NASA 2                               Stennis Space Center   
424        424     NASA 2                       Wallops Flight Facility/GSFC   
425        425     NASA 2                       Wallops Flight Facility/GSFC   
426        426     NASA 2                       Wallops Flight Facility/GSFC   
427        427     NASA 2                       Wallops Flight Facility/GSFC   
428 